### PR TITLE
Revisited Lesson16

### PR DESCRIPTION
**Pull request by vezeli**
Two fixes:

1. Change type declaration with `newtype` into `data` since `AuthorName'` has two fields and `newtype` works only in types with one field

2. `Book''` is missing `AuthorName'` field because it is using `AuthorName'` as constructor this is now fixed

### CODE REVIEW

**Comment by Rhywun** [comment]
I haven't looked at Haskell in a while; I will review when I get a chance. Thank you!

**Comment by vezeli** [comment]
Hey @Rhywun! This is a bit unrelated to the PR but I have seen that you covered both books "Get programming in Haskell" and "Programming in Haskell" (you have repositories with solutions to exercises from the books). Nice work!

Would you be willing to tell me you opinion about them? Which one would you recommend? What are the strengthens of each book individually according to you? Any recommendations which chapters you liked the most?

**Comment by Rhywun** [comment]
@vezeli - I think they're both very good books, probably my favorites of the Haskell books I've looked at for teaching myself. The "Get Programming" book is a little newer and more geared to beginners - I found it very well-written, especially the chapters on functors, applicatives, and monads. He has a unique way of describing them that I found the easiest to understand of anyone's attempts. There are lot of practical examples too. Including a sort of whirlwind tour of popular libraries.

The other book is also well-written but much faster paced and "academic". The examples reflect this - I kind of lost interest towards the end (though I want to return to it some day and finish the book). But it reads like college textbook to me, very clear and precise.

In short I would recommend the "Get Programming" book first but they're both worth your time. Good luck

PS. Both books have exercises which is a huge plus.

**Comment by vezeli** [comment]
Thanks so much for the elaborate answer! :+1:
